generic_tracers/generic_COBALT_diagyaml2register_diag_body.py

```diff
@@ -0,0 +1,110 @@
+"""
+File: 
+    generic_COBALT_diagyaml2register_diag_body.py
+Author: 
+    Kelly Kearney
+Date: 
+    2024/03/27
+Description: 
+    This script builds the Fortran source code that makes up the primary body of the 
+    generic_COBALT_register_diag subroutine.  This routine allows for easy maintenance of the 
+    highly-repetitive and lengthy code.  The resulting generic_COBALT_register_diag_body.f90 is 
+    added to the primary generic_COBALT.f90 module via an include statement, and can be quickly 
+    regenerated when new diagnostics are added or the specific commands in question need to be 
+    modified.
+Usage:
+    python generic_COBALT_diagyaml2register_diag_body.py formatopt
+
+    formatopt is a string indicating the code format to use:
+        original:   The format in the original generic_COBALT.  A (temporary) vardesc-formatted 
+                    variable is created for each diag, then register_diag_field is called using 
+                    fields from that, using pass-by-position syntax for the first 6 inputs and 
+                    name/value for any additional (default if script called w/o input)
+        novardesc:  This format eliminates the creation of an intermediate vardesc variable 
+"""
+
+import yaml
+import sys
+
+# Parse and check input
+
+if len(sys.argv) < 1:
+    formatopt = "original"
+else:
+    formatopt = sys.argv[1]
+
+if not formatopt in ["original", "novardesc"]:
+    raise Exception("Unrecognized format option")
+
+# Read diagnostics from yaml file
+
+yamlfile = 'generic_COBALT_diag.yaml'
+with open(yamlfile, 'r') as f:
+    diag = yaml.safe_load(f)
+    
+    
+# List of primary parameter names.  These include the vardesc type fields and the inputs to
+# register_diag_field that were passed by position (rather than name/value) in the original
+# generic_COBALT code.  
+
+primary = ["name", "longname", "hor_grid", "z_grid", "t_grid", "units", "mem_size", 
+               "var_to_register", "axes"]
+
+# Open file for writing
+
+f = open("generic_COBALT_register_diag_body.f90", "w")
```

Comment:
As @andrew-c-ross pointed out, the current make script (as well as the GFDL FRE workflow) will try to compile files with extensions like *.F90 and *.f90. For now, I suggest we simply change the file extension to `.inc`, so it becomes `generic_COBALT_register_diag_body.inc`. I also quickly checked the MOM6 side, and it looks like they have mostly abandoned the INCLUDE statement except for some header files. I do not think our changes here will have any impact on them, though.

Reply:
I agree, the only INCLUDE statement I am used to seeing is for MOM_memory.

However, MOM6 does use modules and has many statements like 
`use cobalt_register_diag_body  only : register_diag_body`
I am not sure what the motivation for having all of cobalt as one module is, but this seems like a case where having multiple .f90 files with separate modules could benefit the main code. Breaking up the generic_COBALT code into multiple modules would be part of a bigger conversation.

Reply:
Upon further consideration, I think using a separate module, as @theresa-morrison  suggested, would be a better idea than using an INCLUDE statement.